tests/components/roborock/test_config_flow.py

```diff
@@ -178,3 +183,47 @@ async def test_config_flow_failures_code_login(
     assert result["data"] == MOCK_CONFIG
     assert result["result"]
     assert len(mock_setup.mock_calls) == 1
+
+
+async def test_reauth_flow(
+    hass: HomeAssistant, bypass_api_fixture, mock_roborock_entry: MockConfigEntry
+) -> None:
+    """Test reauth flow."""
+    # Fail authenticating on an already created entry.
+    with patch(
+        "homeassistant.components.roborock.RoborockApiClient.get_home_data",
+        side_effect=RoborockInvalidCredentials(),
+    ):
+        await async_setup_component(hass, DOMAIN, {})
+        assert mock_roborock_entry.state is ConfigEntryState.SETUP_ERROR
+    # Start reauth
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={
+            "source": config_entries.SOURCE_REAUTH,
+            "unique_id": mock_roborock_entry.unique_id,
+            "entry_id": mock_roborock_entry.entry_id,
+        },
+        data={CONF_USERNAME: USER_EMAIL},
+    )
+    assert result["step_id"] == "reauth_confirm"
+    assert result["type"] == FlowResultType.FORM
+    # Request a new code
+    with patch(
+        "homeassistant.components.roborock.config_flow.RoborockApiClient.request_code"
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input={}
+        )
+    # Enter a new code
+    assert result["step_id"] == "code"
+    assert result["type"] == FlowResultType.FORM
+    with patch(
+        "homeassistant.components.roborock.config_flow.RoborockApiClient.code_login",
+        return_value=USER_DATA,
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input={CONF_ENTRY_CODE: "123456"}
+        )
+    assert result["type"] == FlowResultType.ABORT
+    assert result["reason"] == "reauth_successful"
```

Comment:
Are there details of the config entry that need to be verified that they were updated with new values and/or preserved from the existing entry?